liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java

```diff
@@ -0,0 +1,494 @@
+            return classesDirAbsPath;
+        }
+        return null; // directory does not exist.
+    }
+
+    public String getEEVersion(MavenProject project) {
+        List<Dependency> dependencies = project.getDependencies();
+        for (Dependency d : dependencies) {
+            if (!d.getScope().equals("provided")) {
+                continue;
+            }
+            log.debug("getEEVersion, dep="+d.getGroupId()+":"+d.getArtifactId()+":"+d.getVersion());
+            if (d.getGroupId().equals("io.openliberty.features")) {
+                String id = d.getArtifactId();
+                if (id.equals("javaee-7.0")) {
+                    return "ee7";
+                } else if (id.equals("javaee-8.0")) {
+                    return "ee8";
+                } else if (id.equals("javaeeClient-7.0")) {
+                    return "ee7";
+                } else if (id.equals("javaeeClient-8.0")) {
+                    return "ee8";
+                } else if (id.equals("jakartaee-8.0")) {
+                    return "ee8";
+                }
+            } else if (d.getGroupId().equals("jakarta.platform") &&
+                    d.getArtifactId().equals("jakarta.jakartaee-api") &&
+                    d.getVersion().equals("8.0.0")) {
+                return "ee8";
+            }
+        }
+        return null;
+    }
+
+    public String getMPVersion(MavenProject project) {  // figure out correct level of mp from declared dependencies
+        List<Dependency> dependencies = project.getDependencies();
+        for (Dependency d : dependencies) {
+            if (!d.getScope().equals("provided")) {
+                continue;
+            }
+            if (d.getGroupId().equals("org.eclipse.microprofile") &&
+                d.getArtifactId().equals("microprofile")) {
+                String version = d.getVersion();
+                log.debug("dep=org.eclipse.microprofile:microprofile version="+version);
+                if (version.startsWith("1")) {
+                    return "mp1";
+                } else if (version.startsWith("2")) {
+                    return "mp2";
+                } else if (version.startsWith("3")) {
+                    return "mp3";
+                }
+                return "mp4"; // add support for future versions of MicroProfile here
+            }
+            // if (d.getGroupId().equals("io.openliberty.features")) {
+            //     mpVersion = Math.max(mpVersion, getMPVersion(d.getArtifactId()));
+            //     log.debug("dep=io.openliberty.features:"+d.getArtifactId()+" mpVersion="+mpVersion);
+            // }
+        }
+        return null;
+    }
+
+    // logic to map individual deps to MP versions, not currently in use
+    public static int getMPVersion(String shortName) {
+        final int MP_VERSIONS = 4; // number of version columns in table
+        String[][] mpComponents = {
+            // Name, MP1 version, MP2 version, MP3 version
```

Comment:
Looks like MP4 version is included also?

Reply:
I'll fix this.